### PR TITLE
Update textual_inversion.py : persistent_workers=True

### PR DESCRIPTION
**Pull request by uminaty**
I noticed that for small datasets (~10 images), leaving persistent_workers=False as it is by default slowed down training considerably.

By the way, maybe we can change this for all training examples involving small datasets (like LoRA) ? or at least set it as an argument?

- Training examples: @sayakpaul and @patrickvonplaten

### CODE REVIEW

**Comment by sayakpaul** [comment]
@patil-suraj WDYT?

**Comment by bghira** [comment]
it might be better to have an argument to disable persistent workers, and enable it by default.

the problem with this is for larger datasets, the memory isn't released at the end of each epoch. but for particularly small datasets that roll over to the next epoch frequently, this would help quite a lot.

if you are relying on the random states for dataset shuffling, then enabling persistent workers can result in the same data being read in sequence on each epoch. this is because they maintain their state as a part of their process.

if you're using more than one dataloader process, hitting ^C (ctrl-c) to terminate the trainer might leave them lingering in the background. this is truly annoying, but doing `kill $(pidof python)` is dangerous workaround.

as well, if you plug an IterableDataset into this, it can't use persistent workers.

**Comment by uminaty** [comment]
If I understand correctly, the presistent_worker argument should not interfere with the shullfe argument in the dataloader ([source?](https://discuss.pytorch.org/t/what-are-the-dis-advantages-of-persistent-workers/102110). Is this what you meant?

Anyway, I agree it should be set as an argument. Should I create a PR for LoRA and textual inversion training exemples ?

**Comment by bghira** [comment]
it just changes the behaviour, such that previously reproducible training will probably no longer be

**Comment by bghira** [comment]
might also be worth enabling dataset padding if they are split across processes to ensure all processes have same number

**Review by sayakpaul — APPROVED** [review]

**Comment by sayakpaul** [comment]
@uminaty could you run `make style && make quality` on the PR?

**Comment by uminaty** [comment]
@sayakpaul Do you want me to add an argument `persistent_workers` instead in textual inversion and LoRA training examples (set to True by default) ? 

btw I just ran `make style && make quality`
